tests/test_tools.py

```diff
@@ -55,6 +55,44 @@ async def invalid_tool(ctx: RunContext[None]) -> str:  # pragma: no cover
     )
 
 
+def test_builtin_tool_registry():
+    """
+    Test that built-in functions can be registered as tools.
+    """
+    from pydantic_ai import Agent
+    from pydantic_ai.exceptions import UserError
+    from pydantic_ai.models.test import TestModel
+
+    with pytest.raises(UserError):
+        Agent(TestModel(), tools=[max])
+
+    with pytest.raises(UserError):
+        agent = Agent(TestModel())
+        agent.tool_plain(min)
+
+
+def test_builtin_tool_raises_signature_error():
```

Comment:
Can this directly test the function you edited through Pydantic AI calls, instead of by replicating all the calls here?